### PR TITLE
[System] When doing process redirect, use Console.OutputEncoding instead of Console.Out.Encoding.

### LINKED ISSUE #8417: Redirecting process output when Console.SetOut set gives strange output
## Steps to Reproduce

1. Download https://gist.github.com/chamons/8aed519dbc9fb32452c91caa2f2f91e9 as example.cs
2. csc example.cs && mono example.exe
3. Not the differences between the echo command and what you get back

## Current Behavior

        /bin/echo who am i
Captured:桷⁯     浡椠         

## Expected Behavior

The behavior when USE_REDIRECT is commented out up above:

	/bin/echo who am i
Captured:who am i

### On which platforms did you notice this

[X] macOS
[ ] Linux
[ ] Windows

**Version Used**:

 $ mono --version
Mono JIT compiler version 5.10.1.38 (2017-12/b4e428d7c41 Thu Apr  5 19:49:28 EDT 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(3.6.0svn-mono-master/8b1520c8aae)
	GC:            sgen (concurrent by default)

### PR DESCRIPTION
**Pull request by kumpera**
This change makes mono behave in the same way as dotnet, which uses Console.OutputEncoding instead of whatever Console.Out is.

Fixes https://github.com/mono/mono/issues/8417.

### CODE REVIEW

File: mono/tests/bug-8417.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#define USE_REDIRECT
```
Comment (luhenry) [inline mono/tests/bug-8417.cs]: This test should probably be added to https://github.com/mono/mono/blob/master/mcs/class/System/Test/System.Diagnostics/ProcessTest.cs instead

**Comment by luhenry** [comment]
@monojenkins build Linux x64

**Comment by akoeplinger** [comment]
should we revert the changes from https://github.com/mono/mono/pull/3730 too?

**Comment by luhenry** [comment]
@monojenkins build Linux ARMv5

**Comment by luhenry** [comment]
@monojenkins build Linux x64

**Comment by luhenry** [comment]
@monojenkins build Linux x64 Acceptance Tests

**Comment by luhenry** [comment]
@monojenkins build Linux x64 Hybrid Suspend

**Comment by marek-safar** [comment]
@monojenkins backport 2018-04

**Pull request merged**